src/overlays/actors/ovl_En_Vb_Ball/z_en_vb_ball.c

```diff
@@ -27,17 +38,268 @@ const ActorInit En_Vb_Ball_InitVars = {
     (ActorFunc)EnVbBall_Update,
     (ActorFunc)EnVbBall_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Vb_Ball/EnVbBall_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Vb_Ball/EnVbBall_Destroy.s")
+static ColliderCylinderInit sCylinderInit = {
+    { 0x0A, 0x11, 0x09, 0x39, 0x10, 0x01 },
+    { 0x06, { 0x00100700, 0x00, 0x20 }, { 0x00100700, 0x00, 0x00 }, 0x01, 0x01, 0x01 },
+    { 0x0014, 0x0001E, 0x000A, 0x0000, 0x0000, 0x0000 }
+};
+
+void EnVbBall_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnVbBall* this = THIS;
+    ColliderCylinder* collider = &this->collider1;
+    f32 pad;
+    f32 angle;
+
+    if (this->actor.params >= 0xC8) { // Volvagia's bones
+        this->yRotVel = Math_Rand_CenteredFloat(768.0f);
+        this->xRotVel = Math_Rand_CenteredFloat(768.0f);
+        angle = Math_atan2f(this->actor.posRot.pos.x, this->actor.posRot.pos.z);
+        this->actor.velocity.y = Math_Rand_ZeroFloat(3.0f);
+        this->actor.velocity.x = 2.0f * sinf(angle);
+        this->actor.velocity.z = 2.0f * cosf(angle);
+        this->actor.gravity = -0.8f;
+    } else { // Volvagia's rocks
+        Collider_InitCylinder(globalCtx, collider);
+        Collider_SetCylinder(globalCtx, collider, &this->actor, &sCylinderInit);
+        Actor_SetScale(&this->actor, this->actor.posRot.rot.z / 10000.0f);
+        this->collider1.dim.radius = this->actor.scale.y * 3000.0f;
+        this->collider1.dim.height = this->actor.scale.y * 5000.0f;
+        this->collider1.dim.yShift = this->actor.scale.y * -2500.0f;
+        this->xRotVel = Math_Rand_CenteredFloat(8192.0f);
+        this->yRotVel = Math_Rand_CenteredFloat(8192.0f);
+        this->shadowSize = this->actor.scale.y * 68.0f;
+    }
+}
+
+void EnVbBall_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnVbBall* this = THIS;
+    if (this->actor.params < 0xC8) {
+        Collider_DestroyCylinder(globalCtx, &this->collider1);
+    }
+}
+
+void EnVbBall_SpawnDebris(GlobalContext* globalCtx, BossFdParticle* particle, Vec3f* position, Vec3f* velocity,
+                          Vec3f* acceleration, f32 scale) {
+    s16 i1;
```

Comment:
I think for loop iterators should just be called `i` unless there are multiple in the same scope that need different vars (which doesn't seem to be the case here)